library/common/jni_interface.cc

```diff
@@ -52,20 +51,14 @@ extern "C" JNIEXPORT jint JNICALL
 Java_io_envoyproxy_envoymobile_engine_AndroidJniLibrary_initialize(JNIEnv* env,
                                                                    jclass, // class
                                                                    jobject connectivity_manager) {
-  // See note above about c-ares.
-  // c-ares jvm init is necessary in order to let c-ares perform DNS resolution in Envoy.
-  // More information can be found at:
-  // https://c-ares.haxx.se/ares_library_init_android.html
-  ares_library_init_jvm(static_jvm);
-
-  return ares_library_init_android(connectivity_manager);
+  return init_jvm(static_jvm, connectivity_manager);
```

Comment:
With the split implementations, should we move this back to a common JNI layer and have Android call it with a`ConnectivityManager` and Java call it with `null`?

Reply:
The java side doesn't need to call this right now and the Android JNI layer does all this on load (and we'll have to avoid calling the java version of this on the Java JNI layer on load). It feels a bit more brittle since we'll need to remember that we don't want to call this method in the java side because it'll impact the Android side